builders/integrations/relayers/gelato.md

```diff
@@ -36,278 +36,265 @@ Then, take the following steps:
+        _transferRelayFee();
+
+        emit LogBalance(address(this).balance);
+
     }
 }
 ```
 
-Next, you'll create a new test account that will submit the gasless transaction. This account is insecure and should not be used in production. This example defines a `test_token` with a default value for demonstration purposes, but you can specify any token here that you'd like.
+`sendToFriend` takes as arguments the address of a friend, the token contract address, and the amount one would like to send. Notice that `sendToFriend` uses the `onlyGelatoRelay` modifier, inherited from `GelatoRelayContext`, which is very important! It means that only the Gelato Relay smart contract can call your function (`0xaBcC9b596420A9E9172FD5938620E265a0f9Df92` on all supported networks!) `_transferRelayFee()` is a function inherited from `GelatoRelayContext` which allows you to pay Gelato directly during the call forward.
 
-```
-  const test_token = "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE";
-  // Create mock wallet
-  const wallet = Wallet.createRandom();
-  const sponsor = await wallet.getAddress();
-  console.log(`Mock PK: ${await wallet._signingKey().privateKey}`);
-  console.log(`Mock wallet address: ${sponsor}`);
-```
+Similarly, `balanceOf` is a very simple function, which takes no arguments, and emits an event with the current balance of the wallet. Let's try calling `balanceOf` on the [`myDummyWallet` contract](https://moonbeam.moonscan.io/address/0xA045eb75e78f4988d42c3cd201365bDD5D76D406) using Gelato Relay SDK!
 
-### Add Request Data {: #add-request-data }
+### `balanceOf` Typescript Example
 
-In this step you have to provide the ABI-encoded call data for the function you want to interact with. You can generate this by taking the following steps:
+*Please find all Gelato Relay code examples with tests and documentation [here](https://github.com/gelatodigital/relay-docs-examples).* 
 
-1. Navigate to the **Write Contract** heading of the [`HelloWorld.sol` contract on Moonscan](https://moonscan.io/address/0x3456E168d2D7271847808463D6D383D079Bd5Eaa#writeContract){target=_blank} 
-2. Press  **Connect to Web3**. After you accept the terms and conditions, you can connect your wallet
-3. Head to the `sayHiVanilla` function and provide the following default value for the `_feeToken` parameter: `0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE`
-4. Press **Write**
-5. Without confirming the transaction in MetaMask, click on the **Hex** tab
-6. Press **Copy Raw Transaction Data**
+Below is a javascript code example `balanceOf.ts`, which uses Gelato Relay SDK to call `balanceOf` on [`myDummyWallet`](https://moonbeam.moonscan.io/address/0xA045eb75e78f4988d42c3cd201365bDD5D76D406).
 
-The resulting ABI-encodeded call data should look like `0x4b327067000000000000000000000000eeeeeeeeeeeeeeeeeeeeeeeeaeeeeeeeeeeeeeeeee`
+```typescript
+import { GelatoRelaySDK } from "@gelatonetwork/relay-sdk";
+import { ethers, BytesLike } from "ethers";
```

Comment:
Previously we had instructions for the user to install ethers, I think that should be added back in as it's all of a sudden being used here now